plugins/org.jkiss.dbeaver.ext.mysql.ui/src/org/jkiss/dbeaver/ext/mysql/ui/SecondaryStepAuthDialogProvider.java

```diff
@@ -0,0 +1,78 @@
+package org.jkiss.dbeaver.ext.mysql.ui;
+
+import org.eclipse.jface.dialogs.IDialogConstants;
+import org.eclipse.osgi.util.NLS;
+import org.eclipse.swt.SWT;
+import org.eclipse.swt.layout.GridData;
+import org.eclipse.swt.widgets.Composite;
+import org.eclipse.swt.widgets.Shell;
+import org.eclipse.swt.widgets.Text;
+import org.jkiss.dbeaver.ext.mysql.SecondaryStepAuthProvider;
+import org.jkiss.dbeaver.ext.mysql.ui.internal.MySQLUIMessages;
+import org.jkiss.dbeaver.model.runtime.RunnableWithResult;
+import org.jkiss.dbeaver.ui.UIUtils;
+import org.jkiss.dbeaver.ui.dialogs.BaseDialog;
+
+public class SecondaryStepAuthDialogProvider implements SecondaryStepAuthProvider {
```

Comment:
We already have a set of methods for retrieving user credentials.

Please see `DBPPlatformUI#promptUserCredentials`

Reply:
Unfortunately, it's not fully suitable for this case. The option `showSavePassword` there only enables/disables checkbox instead of removing it at all. In this case we can't have Save password option at all. 

![image](https://user-images.githubusercontent.com/28875055/156866311-e342e1a4-cb39-4a73-8f37-38e083a78c0a.png)

Besides that, the label and the checkbox are in a one line while dialog window is not resizable which doesn't work for this case when user opens SQL Script with a long name.

![image](https://user-images.githubusercontent.com/28875055/156866608-e001de8c-c378-40fa-873c-2f3be544f71f.png)